sdk/deviceregistry/azure-resourcemanager-deviceregistry/src/test/java/com/azure/resourcemanager/deviceregistry/DeviceRegistryManagerTests.java

```diff
@@ -0,0 +1,109 @@
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        String registrationState
+                = resourceManager.providers().getByName("Microsoft.DeviceRegistry").registrationState();
+        if (!"Registered".equals(registrationState) && !"registering".equals(registrationState)) {
+            registrationState = resourceManager.providers().register("Microsoft.DeviceRegistry").registrationState();
+        }
+        while (!"Registered".equals(registrationState) && !"registering".equals(registrationState)) {
+            try {
+                Thread.sleep(5000);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            registrationState = resourceManager.providers().getByName("Microsoft.DeviceRegistry").registrationState();
+        }
+
+        deviceRegistryManager = DeviceRegistryManager.configure()
+            .withPolicy(new ProviderRegistrationPolicy(resourceManager))
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        // use AZURE_RESOURCE_GROUP_NAME if run in LIVE CI
+        String testResourceGroup = Configuration.getGlobalConfiguration().get("AZURE_RESOURCE_GROUP_NAME");
+        testEnv = !CoreUtils.isNullOrEmpty(testResourceGroup);
+        if (testEnv) {
+            resourceGroupName = testResourceGroup;
+        } else {
+            resourceManager.resourceGroups().define(resourceGroupName).withRegion(REGION).create();
+        }
+    }
+
+    @Override
+    protected void afterTest() {
+        if (!testEnv) {
+            resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+        }
+    }
+
+    @Test
+    @LiveOnly
+    public void testList() {
+        // The DeviceRegistry service must be supported by the service Kubernetes cluster with Azure Arc
+        // and Microsoft.ExtendedLocation, but Kubernetes cluster with Azure Arc can only be created by script.
+        // so only add search test
+        List<Asset> assets = deviceRegistryManager.assets()
+            .listByResourceGroup(resourceGroupName)
+            .stream()
+            .collect(Collectors.toList());
+        Assertions.assertTrue(assets.isEmpty());
+
+        List<AssetEndpointProfile> assetEndpointProfiles = deviceRegistryManager.assetEndpointProfiles()
+            .listByResourceGroup(resourceGroupName)
+            .stream()
+            .collect(Collectors.toList());
+        Assertions.assertTrue(assetEndpointProfiles.isEmpty());
+
+        List<BillingContainer> billingContainers
+            = deviceRegistryManager.billingContainers().list().stream().collect(Collectors.toList());
+        Assertions.assertTrue(billingContainers.isEmpty());
+    }
+
+    private static String randomPadding() {
+        return String.format("%05d", Math.abs(RANDOM.nextInt() % 100000));
+    }
```

Comment:
Delete if we don't need it.

Reply:
fixed by [c2c10b6](https://github.com/Azure/azure-sdk-for-java/pull/44383/commits/c2c10b6b23d54ea1821b3ab17b8a3a79461e3e55)